### PR TITLE
Fix #5122: Add wallet params to DAU ping

### LINKED ISSUE #5122: Add wallet usage to stats updater
### Description: 

DAU ping should include a new param that includes some wallet data.

This will require some brave-core exposure however as there is no method in KeyringService to get the last unlock date

See: 
- https://github.com/brave/brave-core/blob/master/browser/brave_stats/brave_stats_updater_params.cc
- https://github.com/brave/brave-core/tree/master/components/brave_stats/browser

### Test Plan

- Verify `wallet2` query parameter exists when testing DAU ping (via proxy or working with stats team)
- Verify if wallet is not created, `wallet2` is always `0`
- Otherwise if wallet is created the value is based on last ping and last unlock date, see documentation of this helper method for more details: https://github.com/brave/brave-core/blob/a0bb8c80a1fafd2046ba359324ebd74a2b520dd9/components/brave_stats/browser/brave_stats_updater_util.cc#L107

### PR DESCRIPTION
**Pull request by kylehickinson**
## Summary of Changes

This pull request fixes #5122 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveShared/Analytics/DAU.swift [hunk 1]
```diff
@@ -38,12 +39,17 @@ public class DAU {
 
   private static let apiKeyPlistKey = "API_KEY"
   private let apiKey: String?
+  private let braveCore: BraveCoreMain?
 
-  public init(date: Date = Date()) {
+  public init(
+    date: Date = Date(),
+    braveCore: BraveCoreMain?
```
Comment (iccub) [inline BraveShared/Analytics/DAU.swift]: I wonder whether we need entire brave core passed here or only the `braveStats` part?

Comment (kylehickinson) [reply BraveShared/Analytics/DAU.swift]: Good call, we only need the stats part

Comment (kylehickinson) [reply BraveShared/Analytics/DAU.swift]: Updated to use stats object directly

**Review by iccub — APPROVED** [review]
Looks good, this will need some test plan, either coordinate with the Stats team or let the QA to run Charles Proxy to verify the new stat is sent

**Comment by kylehickinson** [comment]
Added some testing details to the issue itself, will work with stats team when this lands in a Dev build

**Pull request merged**